_posts/2024-02-07-#28950.md

```diff
@@ -8,16 +8,56 @@ authors: [instagibbs]
 host: ismaelsadeeq
 status: upcoming
-commit:
+commit: bf6fbea9794db5e8d0adce8ff6182574808cea54
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+
+- Package transactions submitted to the [`submitpackage`](https://bitcoincore.org/en/doc/26.0.0/rpc/rawtransactions/submitpackage/) RPC successfully are accepted to the node's mempool, but broadcasted to peers individually because [Package relay](https://bitcoinops.org/en/topics/package-relay/) is currently a work in progress. Successful submission does not mean the transactions will propagate.
+
+- `maxfeerate` is an optional parameter for the [`sendrawtransaction`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L35) and [`testmempoolaccept`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L104) RPCs. Transactions exceeding this feerate won't be added to mempool and broadcasted to peers, helping prevent unintentional overpayment.
+
+- `maxburnamount` is a similar optional parameter for the `sendrawtransaction` RPC that helps prevent users from accidentally burning funds. Transactions with a data carrier `OP_RETURN` output amount that exceeds `maxburnamount` (default value: 0) are not submitted or broadcasted.
+
+- Upon decoding a transaction in `sendrawtransaction` and `testmempoolaccept` RPCs, `maxburnamount` sanity check is immediately performed.
+
+- In the `sendrawtransaction` RPC, `maxfeerate` sanity check is performed by calculating the transaction size fee at `maxfeerate` fee rate (as [`max_raw_tx_fee`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L89)) which is passed to [`BroadcastTransaction`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/node/transaction.cpp#L33), transactions whose base fee exceeds the `max_raw_tx_fee` are neither broadcasted nor added to the mempool.
+
+- `testmempoolaccept` performs `maxfeerate` sanity check after package processing. It ensures that the individual base fee of all the successfully validated transactions [does not exceed](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L218) the fee of the transaction size at `maxfeerate` fee rate.
+
+- This PR adds optional parameters `maxfeerate` and `maxburnamount` to the [`submitpackage`](https://github.com/bitcoin-core-review-club/bitcoin/blob/bf6fbea9794db5e8d0adce8ff6182574808cea54/src/rpc/mempool.cpp#L816) RPC, enabling similar sanity checks on package transactions.
+
+- After [decoding each package transaction in `submitpackage`](https://github.com/bitcoin/bitcoin/blob/e69796c79c0aa202087a13ba62d9fbcc1c8754d4/src/rpc/mempool.cpp#L814), the transaction outputs are subjected to a `maxburnamount` sanity check.
```

Comment:
Isn't this a duplicate of the point a few lines up?

Reply:
That is explaining how it's performed for `testmempoolaccept` and `sendrawtransaction` RPC?

It's similar because it's basically the same approach